draft-ietf-privacypass-key-consistency.md

```diff
@@ -91,59 +99,97 @@ solutions to this problem.
 
 This document defines the following terms:
 
-Key Consistency and Correctness System (KCCS):
-: A mechanism for providing clients with a consistent view of cryptographic key material within a period of time.
-
 Reliant System:
-: A system that embeds one or more key consistency and correctness systems.
+: A system that embeds one or more key consistency systems.
+
+Key:
+: A cryptographic object used by a reliant system.
+
+Key Identifier (Key ID):
+: A unique identifier for a key.
+
+Key Set:
+: A set of one or more keys.
+
+Key Set Identifier (Set ID):
+: A unique identifier for a key set.
+
+Client:
+: An entity that uses a key in a reliant system.
 
-The KCCS's consistency model is dependent on the implementation and reliant system's threat model.
+Source:
+: An entity that provides key material for use by clients.
 
-# Core Requirements {#reqs}
+The key consistency model is dependent on the implementation and reliant system's threat model.
+
+# Consistency Requirements {#reqs}
 
 Privacy-focused protocols which rely on widely shared public keys typically
-require keys be consistent and correct. Informally, key consistency is the
-requirement that all users who communicate with an entity share the same view
-of the key associated with that entity; key correctness is that the key's
-secret information is controlled by the intended entity and is not known to be
-available to an external attacker.
-
-Some protocols depend on large sets of users with consistent keys for privacy
-reasons. Specifically, all users with a consistent key represent an anonymity
-set wherein each user of the key in that set is indistinguishable from the
-rest. An attacker that can actively cause inconsistent views of keys can
-therefore compromise user privacy.
-
-An attacker that can cause a user to use an incorrect key will likely compromise
-the entire protocol, not just privacy.
-
-Reliant systems must also consider agility when trying to satisfy these requirements. A naive solution to
-ensuring consistent and correct keys is to only use a single, fixed key pair for the entirety of
-the system. Users can then embed this key into software or elsewhere as needed, without any additional
-mechanics or controls to ensure that other users have a different key. However, this solution clearly
+require keys be consistent. Informally, key consistency is the
+requirement that all clients who use a source-provided key in some reliant system
+share the same view of the key. Some protocols depend on large sets of clients
+with consistent keys for privacy reasons. Specifically, all clients with a
+consistent key represent an anonymity set wherein each client of the key in
```

Comment:
Are we selling this short by not mentioning the set of configuration items as a whole?  OHTTP is careful to note that privacy depends on consistency of not just the key, but the entire  client configuration, of which the key is a part.

The key might be central, but how it is identified and how the protocol is executed are details that could expose clients.

Reply:
Yeah, our collective thinking has evolved from keys to configurations. We can address this separately.

Reply:
Filed #31 to track.